File: seava.mod.ad.presenter/src/main/java/ro/seava/ad/presenter/impl/system/model/DbChangeLog_Ds.java
/**
 * Solutii Ecommerce, Automatizare, Validare si Analiza | Seava.ro
 * Copyright: 2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package ro.seava.ad.presenter.impl.system.model;

import java.util.Date;
import ro.seava.ad.domain.impl.system.DbChangeLog;
import ro.seava.j4e.api.annotation.Ds;
import ro.seava.j4e.api.annotation.DsField;
import ro.seava.j4e.api.annotation.SortField;
import ro.seava.j4e.api.model.IModelWithId;
import ro.seava.j4e.presenter.model.AbstractDsModel;

@Ds(entity = DbChangeLog.class, sort = {@SortField(field = DbChangeLog_Ds.f_orderExecuted, desc = true)})
public class DbChangeLog_Ds extends AbstractDsModel<DbChangeLog>
		implements
			IModelWithId {
	public static final String f_id = "id";
	public static final String f_txid = "txid";
	public static final String f_author = "author";
	public static final String f_filename = "filename";
	public static final String f_dateExecuted = "dateExecuted";
	public static final String f_orderExecuted = "orderExecuted";
	public static final String f_md5sum = "md5sum";
	public static final String f_description = "description";
	public static final String f_comments = "comments";
	public static final String f_tag = "tag";
	public static final String f_liquibase = "liquibase";

	@DsField(fetch = false)
	private String id;

	@DsField
	private String txid;

	@DsField
	private String author;

	@DsField
	private String filename;

	@DsField
	private Date dateExecuted;

	@DsField
	private Integer orderExecuted;

	@DsField
	private String md5sum;

	@DsField
	private String description;

	@DsField
	private String comments;

	@DsField
	private String tag;

	@DsField
	private String liquibase;

	public DbChangeLog_Ds() {
		super();
	}

	public DbChangeLog_Ds(DbChangeLog e) {
		super(e);
	}

	public String getId() {
		return this.id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getTxid() {
		return this.txid;
	}

	public void setTxid(String txid) {
		this.txid = txid;
	}

	public String getAuthor() {
		return this.author;
	}

	public void setAuthor(String author) {
		this.author = author;
	}

	public String getFilename() {
		return this.filename;
	}

	public void setFilename(String filename) {
		this.filename = filename;
	}

	public Date getDateExecuted() {
		return this.dateExecuted;
	}

	public void setDateExecuted(Date dateExecuted) {
		this.dateExecuted = dateExecuted;
	}

	public Integer getOrderExecuted() {
		return this.orderExecuted;
	}

	public void setOrderExecuted(Integer orderExecuted) {
		this.orderExecuted = orderExecuted;
	}

	public String getMd5sum() {
		return this.md5sum;
	}

	public void setMd5sum(String md5sum) {
		this.md5sum = md5sum;
	}

	public String getDescription() {
		return this.description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getComments() {
		return this.comments;
	}

	public void setComments(String comments) {
		this.comments = comments;
	}

	public String getTag() {
		return this.tag;
	}

	public void setTag(String tag) {
		this.tag = tag;
	}

	public String getLiquibase() {
		return this.liquibase;
	}

	public void setLiquibase(String liquibase) {
		this.liquibase = liquibase;
	}
}

File: seava.mod.ad.ui.extjs/src/main/resources/webapp/seava/ad/ui/extjs/ds/DbChangeLog_Ds.js
/**
 * Solutii Ecommerce, Automatizare, Validare si Analiza | Seava.ro
 * Copyright: 2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
	
Ext.define("seava.ad.ui.extjs.ds.DbChangeLog_Ds", {
	extend: 'Ext.data.Model',
	
	statics: {
		ALIAS: "ad_DbChangeLog_Ds"
	},
	
	
	fields: [
		{name:"id", type:"string"},
		{name:"txid", type:"string"},
		{name:"author", type:"string"},
		{name:"filename", type:"string"},
		{name:"dateExecuted", type:"date", dateFormat:Main.MODEL_DATE_FORMAT},
		{name:"orderExecuted", type:"int", useNull:true},
		{name:"md5sum", type:"string"},
		{name:"description", type:"string"},
		{name:"comments", type:"string"},
		{name:"tag", type:"string"},
		{name:"liquibase", type:"string"}
	]
});

Main.createFilterModelFromRecordModel({
	fields: [
		{name:"dateExecuted_From",type:"date", dateFormat:Main.MODEL_DATE_FORMAT},
		{name:"dateExecuted_To",type:"date", dateFormat:Main.MODEL_DATE_FORMAT},
		{name:"orderExecuted_From",type:"int", useNull:true},
		{name:"orderExecuted_To",type:"int", useNull:true}
	],
	recordModelFqn: "seava.ad.ui.extjs.ds.DbChangeLog_Ds"
});
